firmware/src/adc.c

```diff
@@ -42,41 +46,18 @@ uint16_t ma_adc2(void)
     return avg_adc2;
 }
 
-float IpanelToFloat(int adcValue) {
-    // Calcula a parte inteira (em volts)
-    int inteiro = (adcValue * 5) / 1023;
-    // Calcula o resto
-    int resto = (adcValue * 5) % 1023;
-    // Combina a parte inteira com a parte decimal
-    float resultado = inteiro + (float)resto / 1023;
-
-    return resultado;
-}
-
-float VpanelToFloat(int adcValue) {
-    // Calcula a parte inteira (em volts)
-    int inteiro = (adcValue * 21) / 1023;
-    // Calcula o resto
-    int resto = (adcValue * 21) % 1023;
-    // Combina a parte inteira com a parte decimal
-    float resultado = inteiro + (float)resto / 1023;
-
-    return resultado;
+float vpanel_to_float(uint16_t vpanel_adc)
+{
+  return 0.020717284 * (float)vpanel_adc - 0.057896889;

```

Comment:
Maybe add this constants in a #define to be easy to find and change in the future